sklearn/ensemble/gradient_boosting.py

```diff
@@ -997,12 +997,16 @@ def fit(self, X, y, sample_weight=None, monitor=None):
         else:
             sample_weight = column_or_1d(sample_weight, warn=True)
 
-        le = LabelEncoder()
-        le.fit(y)
-        y_ = le.transform(y)
-        if np.count_nonzero(np.bincount(y_, weights=sample_weight)) < 2:
-            raise ValueError("y should contain atleast 2 classes after "
-                             "sample_weight trims samples with zero weights.")
+        if not np.issubdtype(y.dtype, np.dtype('float')):
```

Comment:
I don't get why this is relevant

Reply:
Since `BaseGradientBoosting` can only differentiate between a classification and regression problem via checking the *dtype* of `y`. The original issue mentions the problem to be with classification problem not with regression problems.

Reply:
I was thinking of implementing a separate `fit` method for `GradientBoostingClassifier` containing this ValueError check excluding this float *dtype* comparison check.

Reply:
WDYT?

Reply:
float dtype check is not appropriate. Ints can be regression targets, for instance.
Better off using `isinstance(self,...)`, or if you feel it is not going to duplicate much code, yes, you can implement a separate `GBC.fit`

Reply:
Nope, that would mostly be a duplicate code. I've refrained from it.